index.js

```diff
@@ -9,46 +9,76 @@ app.set("view engine", "ejs");
 
 const startLat = 43.839767;
 const startLng = 18.343170;
-
 const stepDistance = 100;
 
-let currentLat = startLat;
-let currentLng = startLng;
-
-let prevLng = currentLng;
-
-const grid = [];
-for (let i = 0; i < 5; i++) {
-    for (let j = 0; j < 5; j++) {
-        grid.push({
-            lat: move_coordinate_on_y_axis(startLat, startLng, i * stepDistance),
-            lng: move_coordinate_on_x_axis(startLat, startLng, j * stepDistance)
-        });
+function generatePlanePath(startLat, startLng, teamName, gridSize) {
+    let currentLat = startLat;
+    let currentLng = startLng;
+    const grid = [];
+    let prevLng = startLng;
+
+    for (let i = 0; i < 5; i++) {
+        for (let j = 0; j < 5; j++) {
+            if (j === 0) {
+                grid.push({
+                    name: teamName,
+                    lat: move_coordinate_on_y_axis(currentLat, prevLng, i * stepDistance),
+                    lng: prevLng,
+                });
+            } else {
+                const lng = move_coordinate_on_x_axis(
+                    currentLat,
+                    currentLng,
+                    j * stepDistance
+                );
+                grid.push({
+                    name: teamName,
+                    lat: move_coordinate_on_y_axis(currentLat, lng, i * stepDistance),
+                    lng: lng,
+                });
+            }
+        }
+
+        prevLng = move_coordinate_on_x_axis(currentLat, currentLng, 5 * stepDistance);
+        currentLng = prevLng;
     }
-   
-    prevLng = move_coordinate_on_x_axis(currentLat, currentLng, 5 * stepDistance);
-    currentLng = prevLng;
+
+    return grid;
 }
 
+const teamOnePath = generatePlanePath(startLat, startLng, "OSTIM", 5);
+const teamTwoPath = generatePlanePath(startLat + 0.01, startLng - 0.01, "STAMBOL", 5);
+const teamThreePath = generatePlanePath(startLat + 0.02, startLng - 0.02, "PBGS", 5);
+
+const teams = [teamOnePath, teamTwoPath, teamThreePath];
+
 app.get("/uav-location", (req, res) => {
-    res.render("index", { grid, currentPoint: grid[0] });
+    res.render("index", { teams });
 });
 
 io.on("connection", async (socket) => {
     console.log("a user connected");
 
-    for (let i = 0; i < grid.length; i++) {
-        currentLat = grid[i].lat;
-        currentLng = grid[i].lng;
-
-        socket.emit("updateCurrentPoint", { lat: currentLat, lng: currentLng });
-
-        await delay(2000);
+    const promises = [];
+
+    for (let j = 0; j < teams.length; j++) {
+        const team = teams[j];
+        const teamName = team[0].name;
+        for (let i = 0; i < team.length; i++) {
+            const currentPoint = team[i];
+            const promise = delay(2000 * i).then(() => {
+                socket.emit("updateCurrentPoints", { team: teamName, point: currentPoint });
+            });
+            promises.push(promise);
+        }
     }
 
+    await Promise.all(promises);
```

Comment:
ovo funkcionise ali je overkill, moras skratiti ovo bolan, previse je, uradi ovo umjesto svega:

```
for(let i = 0; i < teams.length; i++){
    const teamPositions = teams[i];
    socket.emit("updateCurrentPoints", teamPositions);
    await delay(2000)
}
```

onda ces to u svom ```index.ejs``` samo da malo promijenis imena, jer nema poente da ovako razdvajas podatke osim ako zelis da nesto izostavis. mi necemo da izostavimo nista, tako da nema poente da razdvajas

Reply:
ovo funkcionise ali je overkill, uradi ovo:

```
for(let i = 0; i < teams.length; i++){
    const teamPositions = teams[i];
    socket.emit("...", teamPositions);
    await delay(2000)
}
```

onda ces to u svom ```index.ejs``` samo da malo promijenis imena, jer nema poente da ovako razdvajas podatke osim ako zelis da nesto izostavis. mi necemo da izostavimo nista, tako da nema poente da razdvajas